gym-unity/gym_unity/envs/__init__.py

```diff
@@ -120,19 +129,31 @@ def __init__(
             high = np.array([1] * self.group_spec.action_shape)
             self._action_space = spaces.Box(-high, high, dtype=np.float32)
         high = np.array([np.inf] * self._get_vec_obs_size())
-        if self.use_visual:
-            shape = self._get_vis_obs_shape()
-            if uint8_visual:
-                self._observation_space = spaces.Box(
-                    0, 255, dtype=np.uint8, shape=shape
-                )
+        if not self._allow_multiple_obs:
```

Comment:
I think (haven't gone through all the permuations) you can combine some of the single and multiple obs code with something like
```
list_spaces: List[gym.Space] = []
# ... same setup code for multiple obs ...
if self._allow_multiple_obs:
  self._observation_space = spaces.Tuple(list_spaces)
else:
  self._observation_space = list_spaces[0]
```